backends/ref/ceed-ref-operator.c

```diff
@@ -1089,39 +1089,39 @@ int CeedOperatorCreateFDMElementInverse_Ref(CeedOperator op,
                      "FDMElementInverse only supported for tensor "
                      "bases");
   // LCOV_EXCL_STOP
-  CeedScalar *work, *mass, *laplace, *x, *fdm_interp, *lambda;
-  ierr = CeedMalloc(Q_1d*P_1d, &work); CeedChkBackend(ierr);
-  ierr = CeedMalloc(P_1d*P_1d, &mass); CeedChkBackend(ierr);
-  ierr = CeedMalloc(P_1d*P_1d, &laplace); CeedChkBackend(ierr);
-  ierr = CeedMalloc(P_1d*P_1d, &x); CeedChkBackend(ierr);
-  ierr = CeedMalloc(P_1d*P_1d, &fdm_interp); CeedChkBackend(ierr);
-  ierr = CeedMalloc(P_1d, &lambda); CeedChkBackend(ierr);
+  CeedScalar *mass, *laplace, *x, *fdm_interp, *lambda;
+  ierr = CeedCalloc(P_1d*P_1d, &mass); CeedChkBackend(ierr);
+  ierr = CeedCalloc(P_1d*P_1d, &laplace); CeedChkBackend(ierr);
+  ierr = CeedCalloc(P_1d*P_1d, &x); CeedChkBackend(ierr);
+  ierr = CeedCalloc(P_1d*P_1d, &fdm_interp); CeedChkBackend(ierr);
+  ierr = CeedCalloc(P_1d, &lambda); CeedChkBackend(ierr);
   // -- Mass
   const CeedScalar *interp_1d, *grad_1d, *q_weight_1d;
   ierr = CeedBasisGetInterp1D(basis, &interp_1d); CeedChkBackend(ierr);
   ierr = CeedBasisGetGrad1D(basis, &grad_1d); CeedChkBackend(ierr);
   ierr = CeedBasisGetQWeights(basis, &q_weight_1d); CeedChkBackend(ierr);
-  for (CeedInt i=0; i<Q_1d; i++)
-    for (CeedInt j=0; j<P_1d; j++)
-      work[i+j*Q_1d] = interp_1d[i*P_1d+j]*q_weight_1d[i];
-  ierr = CeedMatrixMultiply(ceed, (const CeedScalar *)work,
-                            (const CeedScalar *)interp_1d, mass, P_1d, P_1d, Q_1d);
-  CeedChkBackend(ierr);
+  for (CeedInt i=0; i<P_1d; i++)
+    for (CeedInt j=0; j<P_1d; j++) {
+      CeedScalar sum = 0.0;
+      for (CeedInt k=0; k<Q_1d; k++)
+        sum += interp_1d[k*P_1d+i]*q_weight_1d[k]*interp_1d[k*P_1d+j];
+      mass[i+j*P_1d] = sum;
+    }
```

Comment:
This helped a bit, but there is still a small difference that's resulting in error being ~1000x larger